### PR TITLE
meta(mssql): use ubuntu-20.04 for tests on 2017

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [ ] Have you added new tests to prevent regressions?
- [ ] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [ ] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description of Changes

As per https://github.com/actions/runner-images/issues/10649, ubuntu server 24.04 does not support running MSSQL 2017

### CODE REVIEW

**Comment by WikiRik** [comment]
Should we remove support for MSSQL 2017? Extended EOL is in about 3 years, but the regular EOL was 2 years ago. https://learn.microsoft.com/en-us/lifecycle/products/sql-server-2017

**Comment by lohart13** [comment]
I'm OK with removing 2017.
LMK if you want me to update the PR to reflect this.

**Comment by WikiRik** [comment]
> I'm OK with removing 2017. LMK if you want me to update the PR to reflect this.

That would be great. Thanks

**Comment by WikiRik** [comment]
Once this is merged, can you update the website as well? https://github.com/sequelize/website/blob/main/src/pages/releases.mdx

**Comment by WikiRik** [comment]
Looking at https://hub.docker.com/r/microsoft/mssql-server it seems that 2019 is also built for Ubuntu 20.04. I'd rather not use old images for the latest version, so maybe we need to split the jobs in the CI config so mssql-oldest runs on ubuntu-20.04 (using mssql 2017-latest) and mssql-latest runs on ubuntu-latest (using mssql 2022-latest)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -227,12 +227,12 @@ jobs:
       - run: yarn start-postgres-${{ matrix.postgres-version }}
       - name: Integration Tests
         run: yarn lerna run test-integration --scope=@sequelize/core
-  test-oldest-latest:
+  test-latest:
```
Comment (WikiRik) [inline .github/workflows/ci.yml:230]: This is not really what I had in mind. I was thinking that we just remove mssql from the dialects in test-oldest-latest and create two new jobs; test-mssql-oldest and test-mssql-latest

Comment (lohart13) [reply .github/workflows/ci.yml]: Ok, sure. Updated now

**Review by WikiRik — APPROVED** [review]
Sorry that I was asking different things, but this looks great. Let's hope Microsoft will do something so that mssql 2017 can also run on the ubuntu-latest again in the future so we don't need this anymore

**Pull request merged**